### ISSUE TITLE
Images with exif-orientation are not displayed properly 

### ISSUE THREAD

**Issue by fleigm** [neos/neos-development-collection#3300]
This is the same issue described here #3148. #3147 should have fixed this but I'm still having this problem.
 
Images with exif orientation get flipped on during thumbnail creation.

### Steps to Reproduce

1. Go to the media module
2. Upload an image with exif orientation [example image](https://github.com/recurser/exif-orientation-examples/blob/master/Landscape_6.jpg)
3. Look at the generated thumbnail

### Expected behavior

Thumbnail should have the same orientation as the original image

### Actual behavior

Thumbnail is rotated by 90°

### Affected version
Neos: 7.0

**Comment by tantegerda1** [comment]
TL;DR: activate the PHP `exif` extension

---

I debugged this issue quite long now. As far as I can see, image _metadata reading_ is **independent** from the Imagine driver. I.e., even with

```
Neos:
  Imagine:
    driver: 'Imagick'
```
EXIF metadata are still read by the PHP extension `exif` - or, [if that is not available - not at all](https://github.com/php-imagine/Imagine/blob/9c094846272f75a2ac865986123b707cf66f49ed/src/Image/Metadata/ExifMetadataReader.php#L42)!

I can clearly reproduce your issue with your example image. Activating the PHP extension `exif` solves the problem.

Now, that leaves us with two questions:
- can you confirm the root cause and the solution (i.e. was `exif` unavailable, does it work with `exif` enabled)?
- how should we proceed? Add `ext-exif` to the a composer suggest section? Warn about unavailable `ext-exif` during setup?

I submitted a [pull request at the upstream repository](https://github.com/php-imagine/Imagine/pull/808), but I'm unsure if this brings enough awareness to the users.

**Comment by sorenmalling** [comment]
@fleigm Does the comment from @tantegerda1 help you ?

**Comment by Andyyyyyy** [comment]
I'm having the same issue right now. I'm running ddev with exif already enabled and loaded (according to `php -r 'print_r(get_loaded_extensions());'`) but the image is still rotated by 90 degrees.
I'm also using the Kaleidoscope package, but I doubt thats the problem.

**Comment by tantegerda1** [comment]
> I'm having the same issue right now. I'm running ddev with exif already enabled and loaded (according to `php -r 'print_r(get_loaded_extensions());'`) but the image is still rotated by 90 degrees. I'm also using the Kaleidoscope package, but I doubt thats the problem.

Please don't take offense if I'm asking too basic questions. Have you made sure that:
- the php config actually in use in the problematic situation is the same as where you've shown the exif extension (think fpm vs. web server vs. cli)
- the image _does_ have an exif rotation specified?

**Comment by Andyyyyyy** [comment]
> the php config actually in use in the problematic situation is the same as where you've shown the exif extension (think fpm vs. web server vs. cli)

Dumping `get_loaded_extensions()` inside an action controller also displays the exif extension.

> the image does have an exif rotation specified?

Yep. I also tried the example picture in the first post with the same outcome. 

FWIW this only occurs on a Neos 5.3.4 installation, other 7.2 projects don't have this problem so an upgrade could be a solution for me (or maybe the problem lies somewhere completely different). Since the bugfix from #3147 was merged into 4.3 I guessed it'd be fixed in 5.3 too.

**Comment by hlubek** [comment]
I can confirm that this doesn't work as expected, despite the `exif` extension being loaded.

Our scenario is in the asset editor where an editor wants to set a custom crop on the image.
It seems the `Autorotate` is not applied, since no `ResizeImageAdjustment` is performed in this case, and thus the coordinates are "flipped" compared to the preview in the inspector.

Since multiple calls to `Autorotate->apply()` are not idempotent, we should apply the autorotate filter only once after `$this->imagineService->open($resourceUri)`. I'll create a pull request for this.
